### PR TITLE
fix: converting `Pinecone` metadata fields from float back to int

### PR DESCRIPTION
**Pull request by davidsbatista**
### Proposed Changes:

Pinecone converts all meta numbers in the meta field to float (https://docs.pinecone.io/guides/data/filter-with-metadata). This causes the `SentenceWindowRetriever` to crash completely.

This PR checks if the metadata values are floats and convert them back to integers making `Pinecone` supported by the `SentenceWindowRetriever`

### How did you test it?

- manual verification, unit tests and integration tests

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py
Comment (anakin87) [inline integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py:273]: Let's explain we only do this for specific metadata fields used by some retrievers.
nit: "Pinecone" instead of "PineCone"

File: integrations/pinecone/tests/test_document_store.py
Comment (anakin87) [inline integrations/pinecone/tests/test_document_store.py:264]:
```suggestion
```
this marker is inherited from `TestDocumentStore` test class, so we don't need it.

File: integrations/pinecone/tests/test_document_store.py
Comment (anakin87) [inline integrations/pinecone/tests/test_document_store.py:265]: I believe this test could be made much shorter if we use the `document_store` fixture, as done for example in `test_write_documents_duplicate_overwrite`.

The fixture should take care of creating and deleting the document store, automatically sleep after writing docs, etc...

**Review by anakin87 — COMMENTED** [review]
The logic looks good.

I found some small opportunities for improvement.
Let's also add a small unit test for the new class method.

**Comment by davidsbatista** [comment]
Do you think an extra unit test is necessary? It's a private function, and testing its functionality is already covered by other test

**Comment by anakin87** [comment]
> Do you think an extra unit test is necessary? It's a private function, and testing its functionality is already covered by other test

I'll leave it up to you; I'd like to have it.

File: integrations/pinecone/tests/test_document_store.py
Comment (davidsbatista) [reply integrations/pinecone/tests/test_document_store.py:264]: good point 👍🏽 thanks

**Comment by davidsbatista** [comment]
> > Do you think an extra unit test is necessary? It's a private function, and testing its functionality is already covered by other test
> 
> I'll leave it up to you; I'd like to have it.

File: integrations/pinecone/tests/test_document_store.py
Comment (anakin87) [inline integrations/pinecone/tests/test_document_store.py:264]: You don't need `sleep_time`. It should be automatically applied when using the `PineconeDocumentStore` fixture.
https://github.com/deepset-ai/haystack-core-integrations/pull/1034#discussion_r1735809950

File: integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py
Comment (anakin87) [inline integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py:273]: Let's explain we only do this for specific metadata fields used by some retrievers.

Comment (davidsbatista) [reply integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py:273]: I need your help in understanding what's missing, it's explicitly on the docstring why and how is used, even being a private function.

Please add here, in the comment, the exact wording you would like to see, and I change it.

**Review by anakin87 — APPROVED** [review]
I pushed some small changes.
I'll merge if the tests pass.

**Pull request merged**
